src/Sarif/Map/JsonMapSettings.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Sarif.Map
+{
+    /// <summary>
+    ///  JsonMapSettings contains settings to control how JsonMaps are constructed.
+    ///  For now, this is only the Map size target, to determine how much detail the map will include.
+    /// </summary>
+    public class JsonMapSettings
+    {
+        public const long NodeSizeEstimateBytes = 90;         // Bytes for one Node with containing property but not including children
+        public const long ArrayStartSizeEstimateBytes = 5;    // Bytes for one ArrayStart location, if under 10KB.
+        public const long Megabyte = 1024 * 1024;
+
+        public double CurrentSizeRatio { get; set; }
+        public int MinimumSizeForNode { get; set; }
+
+        public double MapDesiredSizeRatio { get; set; }
+        public double MapMaximumSizeBytes { get; set; }
+
+        /// <summary>
+        ///  Default Settings: Map is 1% of size of file mapped; up to a limit of 10 MB.
+        /// </summary>
+        public static JsonMapSettings DefaultSettings => new JsonMapSettings(0.01, 10 * Megabyte);
+
+        /// <summary>
+        ///  Construct JsonMapSettings for a given target size ratio and size limit.
+        /// </summary>
+        /// <param name="mapSizeRatio">Target Map size (0.01 means map should be 1% of file size or JSON being mapped)</param>
+        /// <param name="mapMaximumSizeBytes">Map size limit, in bytes (10 * 1024 * 1024 means 10 MB), 0 for no limit</param>
```

Comment:
>10 * 1024 * 1024 means 10 MB [](start = 72, length = 28)

Not necessary to say that 10 * 1024 * 1024 = 10 MB

Reply:
Changed to be in terms of constant. I still want to propose a default in the comment and propose a representation of it that someone calling this ctor should consider.